src/class-email-types.php

```diff
@@ -24,22 +24,130 @@ class Email_Types {
 	 * @return void
 	 */
 	public function __construct() {
-		add_action( 'init', [ $this, 'maybe_register_settings_page' ] );
-		add_filter( 'pre_update_option_' . static::SETTINGS_KEY, [ $this, 'sort_settings_on_save' ], 10, 1 );
+		// add_action( 'init', [ $this, 'maybe_register_settings_page' ] );
+		// add_filter( 'pre_update_option_' . static::SETTINGS_KEY, [ $this, 'sort_settings_on_save' ], 10, 1 );
+
+		add_action( 'admin_init', [ $this, 'register_scripts' ] );
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ] );
+		add_action( 'admin_menu', [ $this, 'register_submenu_page' ] );
+		add_action( 'admin_init', [ $this, 'register_settings' ] );
+		add_action( 'rest_api_init', [ $this, 'register_settings' ] );
 	}
 
 	/**
-	 * Registers the submenu settings page for the Email Types.
-	 *
-	 * @return void
+	 * Register scripts for the settings page.
+	 */
+	public function register_scripts() {
+		wp_register_script(
+			'wp-newsletter-builder-admin-email-types',
+			get_entry_asset_url( 'wp-newsletter-builder-admin-email-types' ),
+			array_merge( get_asset_dependency_array( 'wp-newsletter-builder-admin-email-types' ), [ 'wp-editor' ] ),
+			get_asset_version( 'wp-newsletter-builder-admin-email-types' ),
+			true
+		);
+		wp_set_script_translations( 'wp-newsletter-builder-admin-email-types' );
+	}
+
+	/**
+	 * Enqueue scripts and styles for the settings page.
+	 */
+	public function enqueue_assets() {
+		wp_enqueue_script( 'wp-newsletter-builder-admin-email-types' );
```

Comment:
Should this logic only be fired on the page where settings are set vs. on all admin pages?